sim/mage/living_bomb.go

```diff
@@ -29,85 +25,73 @@ func (mage *Mage) registerLivingBombSpell() {
 		ThreatMultiplier: 1 - 0.1*float64(mage.Talents.BurningSoul),
 
 		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
-			baseDamage := 690 + (1.5/3.5)*spell.SpellPower()
+			baseDamage := 690 + 0.4*spell.SpellPower()
 			baseDamage *= sim.Encounter.AOECapMultiplier()
 			for _, aoeTarget := range sim.Encounter.Targets {
 				spell.CalcAndDealDamage(sim, &aoeTarget.Unit, baseDamage, spell.OutcomeMagicHitAndCrit)
 			}
 		},
 	})
 
-	target := mage.CurrentTarget
-	hasGlyphOfLivingBomb := mage.HasMajorGlyph(proto.MageMajorGlyph_GlyphOfLivingBomb)
-
 	mage.LivingBomb = mage.RegisterSpell(core.SpellConfig{
-		ActionID:     actionIDSpell,
+		ActionID:     core.ActionID{SpellID: 55360},
 		SpellSchool:  core.SpellSchoolFire,
-		ProcMask:     core.ProcMaskEmpty,
+		ProcMask:     core.ProcMaskSpellDamage,
 		Flags:        SpellFlagMage,
 		ResourceType: stats.Mana,
 		BaseCost:     baseCost,
 
 		Cast: core.CastConfig{
 			DefaultCast: core.Cast{
 				Cost: baseCost,
-
-				GCD: core.GCDDefault,
+				GCD:  core.GCDDefault,
 			},
 		},
 
+		BonusCritRating:  bonusCrit,
+		DamageMultiplier: mage.spellDamageMultiplier,
+		CritMultiplier:   mage.SpellCritMultiplier(1, mage.bonusCritDamage),
+		ThreatMultiplier: 1 - 0.1*float64(mage.Talents.BurningSoul),
+
 		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
 			result := spell.CalcOutcome(sim, target, spell.OutcomeMagicHit)
-			// TODO: Uncomment this
-			// if result.Landed() {
-			mage.LivingBombDots[mage.CurrentTarget.Index].Apply(sim)
-			// }
+			if result.Landed() {
+				mage.LivingBombDot.Apply(sim)
+			}
 			spell.DealOutcome(sim, result)
 		},
 	})
 
-	livingBombDotSpell := mage.RegisterSpell(core.SpellConfig{
-		ActionID:         actionIDDot,
-		SpellSchool:      core.SpellSchoolFire,
-		ProcMask:         core.ProcMaskEmpty,
-		Flags:            SpellFlagMage,
-		Cast:             core.CastConfig{},
-		BonusCritRating:  bonusCrit,
-		DamageMultiplier: mage.spellDamageMultiplier,
-		CritMultiplier:   mage.SpellCritMultiplier(1, mage.bonusCritDamage),
-		ThreatMultiplier: 1 - 0.1*float64(mage.Talents.BurningSoul),
-	})
+	target := mage.CurrentTarget
 
-	mage.LivingBombDots[target.Index] = core.NewDot(core.Dot{
-		Spell: livingBombDotSpell,
+	onTick := func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
```

Comment:
I haven't tried this yet, but someone added a generic form of the `Ternary` function to our utils (https://github.com/wowsims/wotlk/blob/master/sim/core/utils.go#L130) which would allow you to inline this.

Reply:
I'll try a refactoring run using `Ternary` instead of its non-generic siblings. Probably easier to use, and easier to refactor.